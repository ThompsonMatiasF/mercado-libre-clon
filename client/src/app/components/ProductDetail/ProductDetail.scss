@import "src/styles/custom";

.detail {
  padding: 32px;
  .product {
    width: calc(100% - 210px);
    //min-width: 680px;
    img {
      width: 100%;
      max-width: 680px;
      height: auto;
      border-radius: 4px;
    }
    div {
      margin-top: 150px;
    }
  }
  .buy {
    width: 210px;
    min-width: 210px;
    max-width: 210px;
    padding-left: 32px;
    span {
      padding-bottom: 32px;
      &:nth-child(1) {
        font-size: 14px;
        line-height: 14px;
        padding-bottom: 16px;
      }
      &:nth-child(2) {
        font-size: 24px;
        line-height: 24px;
      }
      &:nth-child(3) {
        font-size: 46px;
        line-height: 46px;
      }
    }
  }
}
