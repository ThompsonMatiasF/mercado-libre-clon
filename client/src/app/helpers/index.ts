export { default as mocks } from './mocks'
