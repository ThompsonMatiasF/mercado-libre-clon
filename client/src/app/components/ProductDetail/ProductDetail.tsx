import React, {FC} from 'react';
import { Button } from 'react-bootstrap';
import './ProductDetail.scss';
import { DetailItem } from 'app/types';

interface ProductDetailProps {
  data: DetailItem
}

export const ProductDetail: FC<ProductDetailProps> = ({ data }) => (
  <div className="detail bg-white">

    <div>
      <div className="d-flex justify-content-between flex-wrap-reverse">
        <div className="product">
          <img src={data.picture} alt="product image" className="" />

          <div>
            <span className="h3">Descripcion del producto</span>
            <p style={{ width: '100%' }}>{ data.description }</p>
          </div>
        </div>
        <div className="buy mb-5">
          <span className="d-block">{`${data.condition} - ${data.sold_quantity} vendidos`}</span>
          <span className="d-block font-weight-bold">{ data.title }</span>
          <span className="d-block">
            $
            {' '}
            {new Intl.NumberFormat('es-AR').format(data.price.amount)}
          </span>
          <Button variant="primary" block>Comprar</Button>
        </div>
      </div>
    </div>
  </div>
);

export default ProductDetail;
