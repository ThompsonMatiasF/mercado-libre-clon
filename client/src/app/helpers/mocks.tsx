import React, { FC } from 'react';
import { AppContext } from 'app/store';

interface MockContextProps {
  context: any
}

export const MockContext: FC<MockContextProps> = ({ context, children })  => (
  <AppContext.Provider value={ context }>{ children }</AppContext.Provider>
)

export const mockCategories = ['Celulares y Teléfonos', 'Celulares y Smartphones']

const mockPrice = {
  currency: 'ARS',
  amount: 100,
  decimals: 0,
}

export const mockItem = {
  id: 'MLA844506827',
  title: 'Iphone X 128gb',
  price: mockPrice,
  picture: 'https://domain.com/image.png',
  condition: 'new',
  free_shipping: true,
  address: 'Capital Federal',
}

export const mockItemDetail = {
  ...mockItem,
  categories: mockCategories,
  sold_quantity: 2,
  description: 'Iphone X 128g, nuevo, en caja cerrada con garantía oficial',
}

export const mockItems = Array(5).fill(mockItem)

export const mockSearchResult = {
  categories: mockCategories,
  items: mockItems,
}

export default {
  MockContext,
  mockItems,
  mockItem,
  mockItemDetail,
  mockSearchResult,
}

