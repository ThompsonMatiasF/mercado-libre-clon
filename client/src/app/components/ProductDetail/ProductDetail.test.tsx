import { render } from '@testing-library/react'
import ProductDetail from './ProductDetail'
import { mockItemDetail } from 'app/helpers/mocks';

describe('ProductDetail.tsx', () => {
  it('Should match snapshot', () => {
    const { container } = render(<ProductDetail data={mockItemDetail} />)

    expect(container).toMatchSnapshot()
  })
})
