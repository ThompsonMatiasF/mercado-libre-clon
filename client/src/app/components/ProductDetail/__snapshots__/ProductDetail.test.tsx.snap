// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail.tsx Should match snapshot 1`] = `
<div>
  <div
    class="detail bg-white"
  >
    <div>
      <div
        class="d-flex justify-content-between flex-wrap-reverse"
      >
        <div
          class="product"
        >
          <img
            alt="product image"
            class=""
            src="https://domain.com/image.png"
          />
          <div>
            <span
              class="h3"
            >
              Descripcion del producto
            </span>
            <p
              style="width: 100%;"
            >
              Iphone X 128g, nuevo, en caja cerrada con garant√≠a oficial
            </p>
          </div>
        </div>
        <div
          class="buy mb-5"
        >
          <span
            class="d-block"
          >
            new - 2 vendidos
          </span>
          <span
            class="d-block font-weight-bold"
          >
            Iphone X 128gb
          </span>
          <span
            class="d-block"
          >
            $
             
            100
          </span>
          <button
            class="btn btn-primary btn-block"
            type="button"
          >
            Comprar
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
